cmd/nvidia-device-plugin/plugin-manager.go

```diff
@@ -46,7 +51,8 @@ func NewPluginManager(config *spec.Config) (manager.Interface, error) {
 
 	cdiHandler, err := cdi.New(
 		cdi.WithDeviceListStrategies(deviceListStrategies),
-		cdi.WithDriverRoot(*config.Flags.Plugin.ContainerDriverRoot),
+		cdi.WithDriverRoot(string(driverRoot)),
+		cdi.WithDevRoot(driverRoot.getDevRoot()),
```

Comment:
Question -- why are we not using the `NVIDIA_DEVICE_ROOT` option here that was added to the device plugin CLI? From what I gather, `driverRoot.getDevRoot()` will always resolve to either `driverRoot` or `/`. What if a custom dev root is configured that is neither of these paths?

Reply:
The `devRoot` from the context of CDI in this case is the dev root in the container. This is determined from the `driverRoot` (in the container) by checking if `{{ .driverRoot }}/dev` exists and using that, otherwise using `/dev`.

Reply:
Got it. Let's keep this as is for now.

One thing to consider in the context of the GPU Operator -- I currently set `DEV_ROOT_CTR_PATH` in the `driver-ready` status file, to indicate the container path of the device root. If it makes sense to do so, we could consider leveraging this here in the device-plugin (and other operands).

Reply:
One thought that I had was that we could set this in the management CDI spec. Will have to check what that would mean though.